### PR TITLE
arxiv_latex_cleaner 0.1.27 (new formula)

### PR DESCRIPTION
**Pull request by tnaumann**
Adds arxiv_latex_cleaner (https://github.com/google-research/arxiv-latex-cleaner).

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/arxiv_latex_cleaner [hunk 1]
```diff
@@ -0,0 +1,37 @@
+class ArxivLatexCleaner < Formula
+  include Language::Python::Virtualenv
+
+  desc "Clean LaTeX code to submit to arXiv"
+  homepage "https://github.com/google-research/arxiv-latex-cleaner"
+  url "https://files.pythonhosted.org/packages/d7/82/b32e1991ee2a8fd0db2f02ba8e135430aaeed9507e8008d5e9c8beec8eff/arxiv_latex_cleaner-0.1.27.tar.gz"
+  sha256 "f347fc6082417316247dca97bb76f15ac677eb1f5e65f091013086a30cac4eda"
+  license "Apache-2.0"
+
+  head "https://github.com/google-research/arxiv-latex-cleaner.git", branch: "main"
```
Comment (carlocab) [inline Formula/arxiv_latex_cleaner:10]:
```suggestion
  license "Apache-2.0"
  head "https://github.com/google-research/arxiv-latex-cleaner.git", branch: "main"
```

File: Formula/arxiv_latex_cleaner [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  end
+
+  resource "pyyaml" do
+    url "https://files.pythonhosted.org/packages/36/2b/61d51a2c4f25ef062ae3f74576b01638bebad5e045f747ff12643df63844/PyYAML-6.0.tar.gz"
+    sha256 "68fb519c14306fec9720a2a5b45bc9f0c8d1b9c72adf45c37baedfcd949c35a2"
+  end
+
+  def install
+    virtualenv_install_with_resources
+    prefix.install libexec/"share"
```
Comment (carlocab) [inline Formula/arxiv_latex_cleaner:31]: Why is this needed?

File: Formula/arxiv_latex_cleaner [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  end
+
+  test do
+    system "#{bin}/arxiv_latex_cleaner", prefix
```
Comment (carlocab) [inline Formula/arxiv_latex_cleaner:35]: What does this test?

File: Formula/arxiv_latex_cleaner [hunk 4]
```diff
@@ -0,0 +1,37 @@
+class ArxivLatexCleaner < Formula
```
Comment (carlocab) [inline Formula/arxiv_latex_cleaner:1]: This file needs to have a `.rb` extension. `brew` (or even GitHub) doesn't recognise it as a formula file otherwise.

File: Formula/arxiv_latex_cleaner [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+  resource "absl_py" do
+    url "https://files.pythonhosted.org/packages/bc/44/3ab719b4fea06882351cd9f9582c15ba5b4d376992ac40c3ed377761a172/absl-py-1.0.0.tar.gz"
+    sha256 "ac511215c01ee9ae47b19716599e8ccfa746f2e18de72bdf641b79b22afa27ea"
+  end
+
+  resource "pillow" do
+    url "https://files.pythonhosted.org/packages/4b/83/090146d7871d90a2643d469c319c1d014e41b315ab5cf0f8b4b6a764ef31/Pillow-9.1.0.tar.gz"
+    sha256 "f401ed2bbb155e1ade150ccc63db1a4f6c1909d3d378f7d1235a44e90d75fb97"
+  end
```
Comment (carlocab) [inline Formula/arxiv_latex_cleaner:22]:
```suggestion
```

Let's do
```ruby
depends_on "pillow"
```
instead.

File: Formula/arxiv_latex_cleaner (hunk 4 above)
Comment (tnaumann) [reply Formula/arxiv_latex_cleaner:1]: Yikes! Yes, it certainly does.

File: Formula/arxiv_latex_cleaner (hunk 2 above)
Comment (tnaumann) [reply Formula/arxiv_latex_cleaner]: Not entirely sure that it is. I had been modeling after other formula. It looks like only the first part (`virtualenv_install_with_resources`) is necessary.

File: Formula/arxiv_latex_cleaner.rb [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  include Language::Python::Virtualenv
+
+  desc "Clean LaTeX code to submit to arXiv"
+  homepage "https://github.com/google-research/arxiv-latex-cleaner"
+  url "https://files.pythonhosted.org/packages/d7/82/b32e1991ee2a8fd0db2f02ba8e135430aaeed9507e8008d5e9c8beec8eff/arxiv_latex_cleaner-0.1.27.tar.gz"
+  sha256 "f347fc6082417316247dca97bb76f15ac677eb1f5e65f091013086a30cac4eda"
+  license "Apache-2.0"
+  head "https://github.com/google-research/arxiv-latex-cleaner.git", branch: "main"
+
+  depends_on "python@3.10"
```
Comment (carlocab) [inline Formula/arxiv_latex_cleaner.rb:11]: Don't you need a `pillow` dependency?

File: Formula/arxiv_latex_cleaner (hunk 3 above)
Comment (tnaumann) [reply Formula/arxiv_latex_cleaner]: Checks to see that utility is available. Perhaps a more meaningful test would be something along the lines of:

Setup:
```
mkdir latex
echo "% remove\nkeep" > latex/test.tex
arxiv_latex_cleaner latex
```

Expected result:
- `latex_arXiv` directory exists
- `latex_arXiv/test.tex` contains only "keep"

It's unclear to me how to write this as an appropriate test. Suggestion would be welcome.

File: Formula/arxiv_latex_cleaner.rb (hunk 6 above)
Comment (tnaumann) [reply Formula/arxiv_latex_cleaner.rb:11]: Yes, added in [e8720b1](https://github.com/Homebrew/homebrew-core/pull/100405/commits/e8720b15eb6c85b3e4b27d0cf361aa989ebb5fdd).

File: Formula/arxiv_latex_cleaner.rb [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    sha256 "68fb519c14306fec9720a2a5b45bc9f0c8d1b9c72adf45c37baedfcd949c35a2"
+  end
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  test do
+    system "#{bin}/arxiv_latex_cleaner", prefix
+  end
```
Comment (carlocab) [inline Formula/arxiv_latex_cleaner.rb:30]:
```suggestion
  test do
    latexdir = testpath/"latex"
    latexdir.mkpath
    (latexdir/"test.tex").write <<~EOS
      % remove
      keep
    EOS
    system bin/"arxiv_latex_cleaner", latexdir
    assert_predicate testpath/"latex_arXiv", :exist?
    assert_equal "keep", (testpath/"latex_arXiv/test.tex").read.strip
  end
```

File: Formula/arxiv_latex_cleaner.rb [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+  include Language::Python::Virtualenv
+
+  desc "Clean LaTeX code to submit to arXiv"
+  homepage "https://github.com/google-research/arxiv-latex-cleaner"
+  url "https://files.pythonhosted.org/packages/d7/82/b32e1991ee2a8fd0db2f02ba8e135430aaeed9507e8008d5e9c8beec8eff/arxiv_latex_cleaner-0.1.27.tar.gz"
+  sha256 "f347fc6082417316247dca97bb76f15ac677eb1f5e65f091013086a30cac4eda"
+  license "Apache-2.0"
+  head "https://github.com/google-research/arxiv-latex-cleaner.git", branch: "main"
+
+  depends_on "pillow"
```
Comment (bayandin) [inline Formula/arxiv_latex_cleaner.rb:11]: We need to add an entry to `pypi_formula_mappings.json` to ignore `pillow` resource for the following updates.
Also, `absl_py` depends on `six` so we need to add `depends_on "six"` here (and ignore `six` in `pypi_formula_mappings.json` as well)
```
$ pipgrip arxiv_latex_cleaner
arxiv-latex-cleaner==0.1.27
absl-py==1.0.0
six==1.16.0
pillow==9.1.0
pyyaml==6.0
```

File: Formula/arxiv_latex_cleaner.rb [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+  depends_on "python@3.9"
```
Comment (bayandin) [inline Formula/arxiv_latex_cleaner.rb:12]: Can we use `python@3.10`?

```suggestion
  depends_on "python@3.10"
```

File: Formula/arxiv_latex_cleaner.rb (hunk 8 above)
Comment (tnaumann) [reply Formula/arxiv_latex_cleaner.rb:11]: Got it. Added `six` in [7a28063](https://github.com/Homebrew/homebrew-core/pull/100405/commits/7a280638b11b1436a4a2cfaf11570a4834ad3996) and `pypi_forumla_mappings.json` in [225c076](https://github.com/Homebrew/homebrew-core/pull/100405/commits/225c076ce4848a6b140b6f732bb634a30255a6a3).

File: Formula/arxiv_latex_cleaner.rb (hunk 9 above)
Comment (tnaumann) [reply Formula/arxiv_latex_cleaner.rb:12]: I had started with this, but adding `depends_on "pillow"` appeared to break. Rolling back to prior version resolves dependency issue. If there's a better way around this, would be great to know.

File: Formula/arxiv_latex_cleaner.rb (hunk 7 above)
Comment (tnaumann) [reply Formula/arxiv_latex_cleaner.rb]: Thank you @carlocab! This is incredibly helpful and I've added in [b652681](https://github.com/Homebrew/homebrew-core/pull/100405/commits/b6526813acc519eb0f5c044bccdda66d8eb04026).

File: Formula/arxiv_latex_cleaner.rb (hunk 9 above)
Comment (bayandin) [reply Formula/arxiv_latex_cleaner.rb:12]: Ah, I see. We have `pillow` -> `libraqm` ->  `harfbuzz` -> `glib` -> `python@3.9` dependency chain for runtime. And it breaks the idea of having `pillow` built with multiple python versions.

But for the current formula, it looks safe to add it to the `audit_exceptions/versioned_dependencies_conflicts_allowlist.json` list and switch to `python@3.10`. What do you think @carlocab?

Comment (carlocab) [reply Formula/arxiv_latex_cleaner.rb:12]: Fine with that, though all the commits will need to be squashed because @BrewTestBot can't squash commits that modify non-formula files.

**Review by carlocab — APPROVED** [review]

**Review by bayandin — APPROVED** [review]
Thanks @tnaumann! Without contributions like yours, it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉
